pkg/agent/npl/k8s/k8s_controller.go

```diff
@@ -0,0 +1,32 @@
+// +build !windows
+
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8s
+
+import (
+	"github.com/vmware-tanzu/antrea/pkg/agent/npl/portcache"
+
+	clientset "k8s.io/client-go/kubernetes"
+)
+
+type Controller struct {
```

Comment:
@tnqn how do you feel about not introducing a workqueue and the current handling of errors by handlers?

Reply:
Agree with it. For the consideration of efficiency and error handling, the actual work could be done in workers that run in parallel:
1. API events can be merged by workqueue: normally there will be several pod events after a pod is created, a controller normally has workers that process its eventual state only.
2. having several workers to consume the workqueue can improve concurrency.
2. iptables operation is easy to fail when kube-proxy has many rules, having a workqueue can give it chances to retry.

You may refer to any controller in antrea or K8s about how to move the actual process from event handler to worker.

Reply:
@monotosh-avi I would say this is the last major change required for this PR. Even though we are not using a SharedInformer here but a dedicated Informer, I feel like switching to a workqueue is necessary given the nature of the work done by the handlers.

Reply:
Sure @antoninbas , @tnqn . I'll make the changes.